### DISCUSSION TITLE
Function parameters, inputs, sensitivities

### DISCUSSION THREAD

**Discussion by DavidMStraub** [pybamm-team/PyBaMM#2694] [General]
Hi,

I think it's great how input parameters and sensitivities can be used for parameter optimization, like [in this notebook](https://github.com/pybamm-team/workshop-notebooks/blob/main/2021-09-workshop/Session%203/sensitivities_and_data_fitting.ipynb).

However, I would sometimes like to get the sensitivities with respect to a quantity that is used inside a function parameter, for instance,

- coefficients parametrizing the OCP as function of stoichiometry
- an Arrhenius activation energy or overall factor of an exchange-current density
- ...

Am I right in assuming that this would require rewriting the model/submodel such that the quantity I want to get the sensitivity for is a proper, scalar `pybamm.Parameter`? For instance, if I wanted to use a function for `"Electrolyte conductivity [S.m-1]"`, say 
```python
def my_kappa_e(c_e, T):
    a * c_e
```
and want the sensitivity w.r.t. `a` (silly example ...), would the best approach be to monkey-patch `LithiumIonParameters.kappa_e_dimensional` to return `
```python
pybamm.Parameter("a") * self.c_e
```
rather than
```python
pybamm.FunctionParameter("Electrolyte conductivity [S.m-1]", inputs)
```
? Am I on the right track?

**Comment by valentinsulzer** [comment]
You're on the right track but it's easier than that, you can use pybamm expressions inside the functions in the parameter set so just do
```python3
def kappa(c_e, T):
    return pybamm.InputParameter("a") * c_e

parameter_values.update({"Electrolyte conductivity [S.m-1]": kappa})
```
